BSP/Tests/ADC.c

```diff
@@ -0,0 +1,390 @@
+// This is all autogenerated code (except for the 2 functions at the end)
+    // https://github.com/STMicroelectronics/STM32CubeL4/blob/master/Projects/NUCLEO-L432KC/Examples/ADC/ADC_DMA_Transfer/Src/main.c
+    // First is the main.h
+    // Second is the main.c 
+
+/* High level overview 
+    * ADC is configured to continuously convert ADC_CHANNEL_9
+    * Resolution is 12 bits
+    * Data is continuously being transfered between Data Register and aADCxConvertedData buffer (by DMA)
+    * Input: GPIO_PIN_4 on GPIOA
+    * LED3 (on board) turns ON when error occurs
+
+    * Main.h is easy to change so that you can run it on different pins 
+    * Using SysTick - be Careful with timer
+    * Have to have certain stuff in certain directories to build properly
+    * Runs on our device
+    * Been tested (by STM?) w/ NUCLEO-L432KC board
+
+    * How to use?
+        - Rebuild all files and flash into memory
+        - Run the example
+        - watch aADCxConvertedData varaible in debugger live watch
+
+    * From Main.c, basically warnings so you dont mess up 
+        /* STM32L4xx HAL library initialization:
+            - Configure the Flash prefetch
+            - Systick timer is configured by default as source of time base, but user 
+                can eventually implement his proper time base source (a general purpose 
+                timer for example or other time source), keeping in mind that Time base 
+                duration should be kept 1ms since PPP_TIMEOUT_VALUEs are defined and 
+                handled in milliseconds basis.
+            - Set NVIC Group Priority to 4
+            - Low Level Initialization
+            */ 
+    */
+
+
+// Main.h stuff:
+/**
+  ******************************************************************************
+  * @file    ADC/ADC_DMA_Transfer/Inc/main.h
+  * @author  MCD Application Team
+  * @brief   Header for main.c module
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2017 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef __MAIN_H
+#define __MAIN_H
+
+/* Includes ------------------------------------------------------------------*/
+#include "stm32l4xx_hal.h"
+#include "stm32l4xx_nucleo_32.h"
```

Comment:
We're not always gonna be hooking up to the Nucleo while testing, meaning we don't need to always have this included.